src/main.ts

```diff
@@ -379,6 +379,28 @@ export default class ObsidianGit extends Plugin {
             }
         });
 
+        this.addCommand({
+            id: "discard-all",
+            name: "CAUTION: Discard all changes",
+            callback: async () => {
+                const repoExists = await this.app.vault.adapter.exists(`${this.settings.basePath}/.git`);
```

Comment:
Instead of checking for an existing repo, better check it via.
```
if (! await this.isAllInitialized()) return false;
```

Reply:
Thanks! Good point.